src/components/NetworkContext.tsx

```diff
@@ -0,0 +1,11 @@
+import { createContext } from 'react';
+
+export type ConnectivityState = {
```

Comment:
Why do we need this? We have 
```javascript
export type NetworkState = {
  isConnected: boolean;
  actionQueue: EnqueuedAction[];
};
```

if we do something like:
```javascript
interface ConnectivityState {
isConnected: boolean,
}

interface NetworkState extends ConnectivityState {
  actionQueue: EnqueuedAction[]
}
```

we would be able to just reuse typing in helpers and components instead of creating this one. Also currently `react-native-offline` is not utilizing the latest version of `react-native-netifo` because in this file `src/components/NetworkConnectivity.tsx` there are deprication warnings.